File: content/examples/other/leetcode/problems/best-time-to-buy-and-sell-stock/example-2/solution.py
def max_profit(prices):
    min_so_far = float('inf')
    best = 0
    for p in prices:
        min_so_far = min(min_so_far, p)
        best = max(best, p - min_so_far)
    return best

if __name__ == '__main__':
    print(max_profit([7,1,5,3,6,4]))

File: content/examples/other/leetcode/problems/best-time-to-buy-and-sell-stock/example-2/solution.js
function maxProfit(prices) {
  let minSoFar = Infinity;
  let best = 0;
  for (const p of prices) {
    minSoFar = Math.min(minSoFar, p);
    best = Math.max(best, p - minSoFar);
  }
  return best;
}

if (require.main === module) {
  console.log(maxProfit([7,1,5,3,6,4]));
}

module.exports = { maxProfit };
